SphereEversion/ToMathlib/Analysis/Calculus.lean

```diff
@@ -48,10 +48,12 @@ theorem fderiv_prod_eq_add {f : E × F → G} {p : E × F} (hf : DifferentiableA
       fderiv 𝕜 (fun z : E × F ↦ f (z.1, p.2)) p + fderiv 𝕜 (fun z : E × F ↦ f (p.1, z.2)) p := by
   have H₁ : fderiv 𝕜 (fun z : E × F ↦ f (z.1, p.2)) p =
       (fderiv 𝕜 f p).comp (.comp (.inl 𝕜 E F) (.fst 𝕜 E F)) :=
-    (hf.hasFDerivAt.comp _ (hasFDerivAt_fst.prod (hasFDerivAt_const _ _))).fderiv
+    (hf.hasFDerivAt.comp p
+      ((hasFDerivAt_fst (𝕜 := 𝕜) (E := E) (F := F)).prod (hasFDerivAt_const p.2 _))).fderiv
```

Comment:
These changes make me slightly sad, but overall, that's pretty mild.

Reply:
Previous mathlib versions were abusing a bug to get some unification information out of nowhere. When the bug was fixed, similar fixes were needed in mathlib, so this is not surprising.